core/src/main/csharp/ch/cyberduck/core/local/SystemLocal.cs

```diff
@@ -83,62 +82,119 @@ public override bool isSymbolicLink()
+                        return null;
                     }
+                    writer.Write(Path.DirectorySeparatorChar);
+                    // allow everything. This _may_ be bad, _but_ Local uses `Paths.get()` thus we don't need to do any more sanitization here.
+                    writer.Write(hostBuffer.ToString());
                 }
-
-                return sb.ToString();
+                else
+                {
+                    // this is something different than "\\" and "//", ignore as not supported
+                    return null;
+                }
+            }
+            else
+            {
+                /// <see cref="System.Char.IsLetter(char)" />
+                var letter = (c | 0x20) - 'a';
+                if (letter < 0 || letter > 25)
+                {
+                    // letter is not in range A to Z.
+                    return null;
+                }
+                // change lowercase letter to uppercase
+                writer.Write((char)(c & ~0x20));
+                if (reader.Peek() != Path.VolumeSeparatorChar)
+                {
+                    // this is something like C\, CX, C/, 
+                    return null;
+                }
+                writer.Write((char)reader.Read());
+                // at X:, following is only a path, thus pass on to shared implementation
             }
+            SanitizePath(reader, writer);
 
-            return path;
-        }
+            return writer.ToString();
 
-        private static string MakeValidFilename(string name)
-        {
-            if (Utils.IsNotBlank(name))
+            static void ApplyRoot(StringReader reader, StringWriter writer)
             {
-                foreach (char c in Path.GetInvalidFileNameChars())
+                if (reader.Peek() is int p && (p == Path.DirectorySeparatorChar || p == Path.AltDirectorySeparatorChar))
                 {
-                    name = name.Replace(c.ToString(), "_");
+                    reader.Read();
                 }
+                writer.Write(Path.DirectorySeparatorChar);
             }
 
-            return name.Trim();
+            static void SanitizePath(StringReader reader, StringWriter writer)
+            {
+                var invalid = Path.GetInvalidFileNameChars();
+                ApplyRoot(reader, writer);
+                StringBuilder segmentBuilder = new();
+                while (reader.Peek() != -1)
+                {
+                    char c = (char)reader.Read();
+                    if (c == Path.AltDirectorySeparatorChar || c == Path.DirectorySeparatorChar)
+                    {
+                        writer.Write(segmentBuilder.ToString().Trim());
+                        writer.Write(Path.DirectorySeparatorChar);
+                        segmentBuilder.Clear();
+                    }
+                    else if (Array.IndexOf(invalid, c) != -1)
+                    {
+                        segmentBuilder.Append('_');
```

Comment:
@ylangisc Still UrlEncode invalid characters? Asof Windows 10 1803 '%'[^ms-docs] is a valid filename character. Thus rendering this invalid on any system which is not Windows 10 1803 (or newer, which is true for this[^commit] as well)
[^ms-docs]: https://learn.microsoft.com/troubleshoot/windows-client/networking/url-encoding-unc-paths-not-url-decoded
[^commit]: b169fcfbbbf2c3a9a2a9ca239dbe34d6db09347c